sfcli/SmartFace.Cli/Infrastructure/ApiImplementation/CamerasRepository.cs

```diff
@@ -1,29 +1,121 @@
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+using ManagementApi;
+using SmartFace.Cli.Common;
 using SmartFace.Cli.Core.ApiAbstraction;
 using SmartFace.Cli.Core.ApiAbstraction.Models;
-using SmartFace.Cli.Infrastructure.ApiClient.Extensions;
-using SmartFace.ODataClient.Default;
-using SmartFace.ODataClient.Extensions;
-using SmartFace.ODataClient.SmartFace.WebApi.Models.Requests;
+using SmartFace.Cli.Core.Domain;
 
 namespace SmartFace.Cli.Infrastructure.ApiImplementation
 {
-    public class CamerasRepository: ICamerasRepository
+    public class CamerasRepository : ICamerasRepository, IDisposable
     {
-        private Container Container { get; }
-        
-        public CamerasRepository(Container container)
+        private readonly HttpClient _httpClient;
+        private readonly SetupClient _setupClient;
+
+        public CamerasRepository(IApiDefinition apiDefinition, HttpClient httpClient)
         {
-            Container = container;
+            _httpClient = httpClient;
+            _setupClient = new SetupClient(apiDefinition.ApiUrl, _httpClient);
         }
-        
-        public CameraModel Create(CreateCameraModel createCameraModel)
+
+        public async Task<Camera> CreateCameraAsync(CameraRequestData cameraRequestData)
         {
-            var streamSourceData = StreamSourceData.CreateStreamSourceData(createCameraModel.ScopeId, createCameraModel.VideoSource, false);
-            var camera = Container.Cameras.Create(streamSourceData);
-            return new CameraModel
+            if (cameraRequestData.Enabled.HasValue && cameraRequestData.Enabled.Value && string.IsNullOrEmpty(cameraRequestData.Source))
```

Comment:
elaborate pls

Reply:
the diff is mixed up, original file was here sfcli/SmartFace.Cli/Core/Domain/Video/VideoProcessorRepository.cs and the check is from there.

replaced the check with cheking Name a Source properties.